gtsfm/densify/interface_patchmatchnet.py

```diff
@@ -0,0 +1,209 @@
+"""Interface class from gtsfmData to patchmatchnetData
+
+Authors: Ren Liu
+"""
+from typing import Any, Dict, List, Tuple
+
+import cv2
+import numpy as np
+from torch.utils.data import Dataset
+
+from gtsfm.common.image import Image
+from gtsfm.common.gtsfm_data import GtsfmData
+from gtsfm.densify.mvs_math import piecewise_gaussian, to_camera_coordinates
+
+
+class PatchmatchNetData(Dataset):
+    """PatchmatchNetData class for Patchmatch Net. It contains the interface from GtsfmData"""
+
+    def __init__(self, images: Dict[int, Image], sfm_result: GtsfmData, num_views: int = 5) -> None:
+        """Initialize method for PatchmatchnetData
+
+        Args:
+            images: input images (H, W, C) to GTSFM
+            sfm_result: sfm results calculated by GTSFM
+            num_views: number of views, containing 1 reference view and (num_views-1) source views
+        """
+
+        assert images is not None and len(images) > 1
+
+        # cache sfm result
+        self.sfm_result = sfm_result
+
+        # Patchmatch Net meta
+        self.num_views = num_views
+        self.num_stages = 4
```

Comment:
prefer to make a private global module var `NUM_PATCHMATCHNET_STAGES = 4`

and here `self.num_stages = NUM_PATCHMATCHNET_STAGES` to prevent magic number 4